Lesson4/src/Homeworck4_1/Computer.java

```diff
@@ -9,52 +9,68 @@
     private int drive;
     private int counter;
     private int condition = 1;
+    private int limitCounter;
 
     public Computer(String processor, int operational, int drive) {
+        Scanner a = new Scanner(System.in);
+        System.out.print("Enter limited counter\t");
+        this.limitCounter = a.nextInt();
         this.processor = processor;
         this.operational = operational;
         this.drive = drive;
-
     }
 
-
-    public void Info() {
-        System.out.println("processor\t" + processor + "\noperational\t" + operational + "\ndrive\t" + drive + "\ncounter\n" + counter);
+    public void info() {
+        System.out.println("processor\t" + processor + "\noperational\t" + operational + "\ndrive\t" + drive + "\ncounter\t" + counter);
     }
 
-    public void Enable() {
-        this.counter++;
-        if (this.condition == 0 || this.counter > 3) {
-            this.counter--;
+    public void enable() {
+        if (this.condition == 0 || this.counter >= this.limitCounter) {
             System.out.println("the computer burned down");
             return;
-
         }
-
         Random random = new Random();
         int b = random.nextInt(2);
+        //      It's string for wimps;
+//                System.out.println(b);
         Scanner scanner = new Scanner(System.in);
         System.out.println("You enable computer enter number 1 or 0");
         int a = scanner.nextInt();
         if (a == b) {
             this.condition = 1;
+            this.counter++;
         } else {
             this.condition = 0;
         }
+        if (this.condition == 0) {
 
+            System.out.println("the computer burned down");
+            return;
+        }
     }
 
-    public void Off() {
+    public void off() {
+        if (this.condition == 0 || this.counter >= this.limitCounter) {
```

Comment:
good, this is much better. I would also move this check to a separate method that returns boolean, and use it here and in turn on, but I'll leave it up to you